src/lib/ws/helper/subscribe-handler.ts

```diff
@@ -14,10 +14,37 @@ export const subscribeWithHandler = (socket: ServerSocket, event: string, method
 		} catch (error: unknown) {
 			const probe = socket.data.probe;
 			const clientIp = probe.ipAddress;
-			const reason = isError(error) ? error.message : 'unknown';
+			const metadata: Record<string, unknown> = {
+				client: { id: socket.id, ip: clientIp },
+				details: 'unknown',
+			};
+
+			if (isError(error)) {
+				metadata['details'] = error.message;
+			}
+
+			if (Joi.isError(error)) {
+				metadata['details'] = formatJoiError(error);
+				metadata['validationInput'] = error._original;
+			}
+
+			logger.info(`Event "${event}" failed to handle`, metadata);
 
-			logger.info(`Event "${event}" failed to handle for (${reason})`, { client: { id: socket.id, ip: clientIp } });
 			logger.debug(`Details:`, error);
 		}
```

Comment:
We should increase the level here to a warning to see those in production (possibly exclude some specific errors if they generate too much noise). That also means we can join those two log calls.